### DISCUSSION TITLE
Define multiple VPN configs?

### DISCUSSION THREAD

**Discussion by DataFran** [qdm12/gluetun#1694] [Q&A] [answered]
Hello!
I got Gluetun running today and successfully routed a container through it.
When I set it up I defined a single VPN config. Is it possible to define more than one, should that one fail? 
Thanks!

**Comment by bnhf** [comment, marked as answer]
@DataFran 

Typically, one would use one of these environment variables to setup failover within a VPN provider:

![screenshot-github com-2023 06 16-13_41_08](https://github.com/qdm12/gluetun/assets/41088895/0b770d1d-6340-4839-b14d-a5bb17d1216e)

If you're looking to use more than one VPN provider, then no there isn't a way to setup failover between providers. If you're using something like Portainer though, it'd be easy enough to setup a second stack that would be ready to go with everything the same except for the VPN provider.

If one provider stopped working for you, you could stop one stack and start another to switch providers. All containers and data directories could be the same, with only the yaml for the Gluetun container different.

**Reply by DataFran** [reply]
Oh, my bad for leaving this out. It's a custom VPN as my specific one is not supported out of the box. So I have OPENVPN_CUSTOM_CONFIG set to a specific .ovpn file from my provider. Can I comma separate a few different .ovpn files there?

**Reply by bnhf** [reply]
@DataFran 

No unfortunately not. And, based on what's in the Wiki, even if you put multiple "remote" directives in the .ovpn file only the first one is used. I'd suggest putting in a request to support your provider directly in the "Issues" section of this project.

In the meantime, you could have a couple of additional custom .ovpn files ready to go with different names. The alternates could be commented out in your stack until you need them, and then stop the stack and switch the #'s around to use an alternate server if your first choice is down or slow.

**Reply by DataFran** [reply]
Got it, thanks for your help and time I really appreciate it.
